frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/OptionListSelector/OptionListSelector.test.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import { OptionListSelector } from './OptionListSelector';
+import { screen, waitForElementToBeRemoved } from '@testing-library/react';
+import { ComponentType } from 'app-shared/types/ComponentType';
+import userEvent from '@testing-library/user-event';
+import { componentMocks } from '../../../../../../../testing/componentMocks';
+import { renderWithProviders, optionListIdsMock } from '../../../../../../../testing/mocks';
+import { textMock } from '@studio/testing/mocks/i18nMock';
+import { createQueryClientMock } from 'app-shared/mocks/queryClientMock';
+import type { FormComponent } from '../../../../../../../types/FormComponent';
+
+// Test data:
+const mockComponent: FormComponent<ComponentType.Dropdown> = componentMocks[ComponentType.Dropdown];
+const optionsIdMock = optionListIdsMock[0];
+mockComponent.optionsId = optionsIdMock;
+
+const handleComponentChangeMock = jest.fn();
+const getOptionListIds = jest
+  .fn()
+  .mockImplementation(() => Promise.resolve<string[]>(optionListIdsMock));
+
+describe('OptionListSelector', () => {
+  it('should render the component', async () => {
+    renderOptionListSelector();
+    await waitForElementToBeRemoved(
+      screen.queryByText(textMock('ux_editor.modal_properties_loading')),
+    );
+
+    expect(screen.getByText(textMock('ux_editor.modal_properties_code_list'))).toBeInTheDocument();
+  });
+
+  it('should call onChange when option list changes', async () => {
+    const user = userEvent.setup();
+    renderOptionListSelector();
+    await waitForElementToBeRemoved(
+      screen.queryByText(textMock('ux_editor.modal_properties_loading')),
+    );
+
+    await user.click(getDropDownButton());
+    const dropdownOption = screen.getByText(optionListIdsMock[0]);
```

Comment:
Maybe this could be wrapped in a new function `getDropdownOption`?
Same for line 58.